### ISSUE TITLE
Error when building project: Could not load type 'Cosmos.Core.Multiboot'

### ISSUE THREAD

**Issue by tb100** [CosmosOS/Cosmos#1451]
#### Area of Cosmos - What area of Cosmos are we dealing with?
Building a project. (Happens on all output types: ISO, VMware, etc)
 
#### Expected Behaviour - What do you think that should happen?
The ISO should build successfully.

#### Actual Behaviour - What unexpectedly happens?
I get the error `Exception: System.TypeLoadException: Could not load type 'Cosmos.Core.Multiboot' from assembly 'Cosmos.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983'`.

#### Reproduction - How did you get this error to appear?
I pressed the Cosmos (Run) button and got the error

#### Version - Were you using the User Kit or Dev Kit? And what User Kit version or Dev Kit commit (Cosmos, IL2CPU, X#)?
I'm using Dev Kit installed and updated yesterday (August 20th).

---
Here is the error from the output tab:

`1>IL2CPU : error : Exception: System.TypeLoadException: Could not load type 'Cosmos.Core.Multiboot' from assembly 'Cosmos.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983'.`

`1>   at System.Reflection.CustomAttribute._GetPropertyOrFieldData(RuntimeModule pModule, Byte** ppBlobStart, Byte* pBlobEnd, String& name, Boolean& bIsProperty, RuntimeType& type, Object& value)`

`1>   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeModule decoratedModule, Int32 decoratedMetadataToken, Int32 pcaCount, RuntimeType attributeFilterType, Boolean mustBeInheritable, IList derivedAttributes, Boolean isDecoratedTargetSecurityTransparent)`

`1>   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeType type, RuntimeType caType, Boolean inherit)`

`1>   at System.RuntimeType.GetCustomAttributes(Type attributeType, Boolean inherit)`

`1>   at System.Attribute.GetCustomAttributes(MemberInfo element, Type type, Boolean inherit)`

`1>   at System.Reflection.CustomAttributeExtensions.GetCustomAttributes[T](MemberInfo element, Boolean inherit)`

`1>   at Cosmos.IL2CPU.PlugManager.FindPlugImpls(IEnumerable 1 assemblies) in C:\source\Cosmos\IL2CPU\source\Cosmos.IL2CPU\PlugManager.cs:line 82`

`1>   at Cosmos.IL2CPU.ILScanner.Execute(MethodBase aStartMethod, IEnumerable 1 plugsAssemblies) in C:\source\Cosmos\IL2CPU\source\Cosmos.IL2CPU\ILScanner.cs:line 218`

`1>   at Cosmos.IL2CPU.CompilerEngine.Execute() in C:\source\Cosmos\IL2CPU\source\Cosmos.IL2CPU\CompilerEngine.cs:line 200`

`1>IL2CPU task took 00:00:00.7961818`

---
I also tried installing the latest User Kit (20200708), and I got the same error.

**Comment by tb100** [comment]
The file does exist. If you need it, the full path is `C:\source\Cosmos\Cosmos\source\Cosmos.Core_Asm\Multiboot\MultibootAsm.cs`.

**Comment by quajak** [comment]
Can you try running the TestRunner in the Cosmos source to see if that works.

**Comment by tb100** [comment]
I'm following the guide for [TestRunners](https://github.com/CosmosOS/Cosmos/wiki/TestRunner), and I found the `DefaultEngineConfiguration.cs` file, but I'm not sure what I need to add or how to run it. 
Also, I tried a demo (`C:\source\Cosmos\Cosmos\Demos\CosmosGraphicSubsystem`), and it ran fine without any errors. I went back to my project and tried to run it, but got the same error as before.

**Comment by quajak** [comment]
Have you tried installing the devkit? That might fix it if its some problem related to installation.

**Comment by tb100** [comment]
I already have DevKit installed. I also ran `update.bat` and `install-VS2019.bat` earlier today. It builds without any errors, runs the installer and then opens up Visual Studio.

**Comment by quajak** [comment]
Can you check if AppData\Roaming\Cosmos User Kit\packages contains a Cosmos.Core nupkg file?

**Comment by tb100** [comment]
Yes, there is a Cosmos.Core file. The full filename is `Cosmos.Core.0.1.0-localbuild20200824022219.nupkg`.

**Comment by quajak** [comment]
Can you look at the Cosmos.Core.dll in AppData\Roaming\Cosmos User Kit\Kernel with ILSpy or some other tool to confirm that it contains the Multiboot class?

**Comment by tb100** [comment]
I used dotPeek by Jet Brains to inspect the `Cosmos.Core.dll` file. I'm pretty sure you mean this:
![image](https://user-images.githubusercontent.com/54515877/91180900-3a040600-e6b6-11ea-9136-a005c86a5bcb.png)

**Comment by tb100** [comment]
I removed the `bin` and `obj` folders and changed `<TargetFramework>netcoreapp2.0</TargetFramework>` to `<TargetFramework>netstandard2.0</TargetFramework>` in `testOS.csproj`. Now I am getting this error: `Exception: System.Exception: Method GetMBIAddress requires a plug, but none is implemented	testOS 1	C:\Users\dev\source\repos\testOS\testOS C:\Users\dev\source\repos\testOS\testOS\IL2CPU	1`.

**Comment by quajak** [comment]
Thats a completely different error now. You are trying to call a method which needs to be implemented in cosmos. Whats the entire build log from VS?

**Comment by tb100** [comment]
Here is the output of the build:
`1>------ Build started: Project: testOS, Configuration: Debug Any CPU ------
1>testOS -> C:\Users\dev\source\repos\testOS\testOS\bin\Debug\netstandard2.0\cosmos\testOS.dll
1>Message: Executing IL2CPU on assembly
1>Message: Kernel Base: Cosmos.System.Kernel
1>Message: Checking target assembly: C:\Users\dev\source\repos\testOS\testOS\bin\Debug\netstandard2.0\cosmos\testOS.dll
1>Detecting fields for type 'IL2CPU.Debug.Symbols.FIELD_INFO'
1>Detecting fields for type 'IL2CPU.Debug.Symbols.FIELD_MAPPING'
1>Detecting fields for type 'IL2CPU.Debug.Symbols.AssemblyFile'
1>Detecting fields for type 'IL2CPU.Debug.Symbols.Document'
1>Detecting fields for type 'IL2CPU.Debug.Symbols.Method'
1>IL2CPU : error : Exception: System.Exception: Method GetMBIAddress requires a plug, but none is implemented
1>   at Cosmos.IL2CPU.ILScanner.Assemble() in C:\Development\Cosmos\IL2CPU\source\Cosmos.IL2CPU\ILScanner.cs:line 988
1>   at Cosmos.IL2CPU.ILScanner.Execute(MethodBase aStartMethod, IEnumerable'1 plugsAssemblies) in C:\Development\Cosmos\IL2CPU\source\Cosmos.IL2CPU\ILScanner.cs:line 266
1>   at Cosmos.IL2CPU.CompilerEngine.Execute() in C:\Development\Cosmos\IL2CPU\source\Cosmos.IL2CPU\CompilerEngine.cs:line 200
1>IL2CPU task took 00:00:04.5546514
1>Done building project "testOS.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========`

**Comment by quajak** [comment]
Should be fixed by #1481

**Comment by tb100** [comment]
Okay! I am running the build now. I will post again when it is done.

**Comment by tb100** [comment]
I updated it using `update.bat` and installed it using `install-VS2019.bat`. I am getting the error `Exception: System.TypeLoadException: Could not load type 'Cosmos.Core.Multiboot' from assembly 'Cosmos.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983` again. I deleted both `bin` and `obj`, and made sure the framework is still `netstandard2.0`, but it still isn't working.

**Comment by quajak** [comment]
Could you please try using https://docs.microsoft.com/en-us/dotnet/framework/tools/fuslogvw-exe-assembly-binding-log-viewer to get more information about the error?

**Comment by tb100** [comment]
I found the location of it on my computer (`C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\FUSLOGVW.exe`), but I can't seem to get it set up. I ran it as an administrator, turned logging on in settings, and even set it to log to a custom path, but when I press the Cosmos (Run) button I get the error, but nothing happens in Fuslogvm.

**Comment by quajak** [comment]
Could you try use https://github.com/awaescher/Fusion ?

**Comment by tb100** [comment]
Okay. I just download an ran the program. I pressed "Record", tried to compile my OS, and got this error in Visual Studio:
`Exception: System.TypeLoadException: Could not load type 'Cosmos.Core.Multiboot' from assembly 'Cosmos.Core, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983'.`
Then, I pressed "Stop". It parsed the session, I scrolled to the error with Cosmos, and got this:
```*** Assembly Binder Log Entry  (2020-10-02 @ 6:51:03 PM) ***

The operation failed.
Bind result: hr = 0x80070002. The system cannot find the file specified.

Assembly manager loaded from:  C:\Windows\Microsoft.NET\Framework\v4.0.30319\clr.dll
Running under executable  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\devenv.exe
--- A detailed error log follows. 

=== Pre-bind state information ===
LOG: DisplayName = Cosmos.VS.ProjectSystem, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
 (Fully-specified)
LOG: Appbase = file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/
LOG: Initial PrivatePath = NULL
LOG: Dynamic Base = NULL
LOG: Cache Base = NULL
LOG: AppName = devenv.exe
Calling assembly : (Unknown).
===
LOG: This bind starts in default load context.
LOG: Using application configuration file: C:\Users\dev\AppData\Local\Microsoft\VisualStudio\16.0_dad5eecf\devenv.exe.config
LOG: Using host configuration file: 
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework\v4.0.30319\config\machine.config.
LOG: Post-policy reference: Cosmos.VS.ProjectSystem, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
LOG: GAC Lookup was unsuccessful.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PublicAssemblies/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PublicAssemblies/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/TestWindow/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/TestWindow/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Platform/Debugger/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Platform/Debugger/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/x86/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/x86/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PublicAssemblies/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PublicAssemblies/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/TestWindow/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/TestWindow/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Platform/Debugger/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Platform/Debugger/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/x86/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/x86/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: All probing URLs attempted and failed.



*** Assembly Binder Log Entry  (2020-10-02 @ 6:51:03 PM) ***

The operation failed.
Bind result: hr = 0x80070002. The system cannot find the file specified.

Assembly manager loaded from:  C:\Windows\Microsoft.NET\Framework\v4.0.30319\clr.dll
Running under executable  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\devenv.exe
--- A detailed error log follows. 

=== Pre-bind state information ===
LOG: DisplayName = Cosmos.VS.ProjectSystem, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
 (Fully-specified)
LOG: Appbase = file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/
LOG: Initial PrivatePath = NULL
LOG: Dynamic Base = NULL
LOG: Cache Base = NULL
LOG: AppName = devenv.exe
Calling assembly : (Unknown).
===
LOG: This bind starts in default load context.
LOG: Using application configuration file: C:\Users\dev\AppData\Local\Microsoft\VisualStudio\16.0_dad5eecf\devenv.exe.config
LOG: Using host configuration file: 
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework\v4.0.30319\config\machine.config.
LOG: Post-policy reference: Cosmos.VS.ProjectSystem, Version=1.0.0.0, Culture=neutral, PublicKeyToken=5ae71220097cb983
LOG: GAC Lookup was unsuccessful.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PublicAssemblies/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PublicAssemblies/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/TestWindow/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/TestWindow/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Platform/Debugger/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Platform/Debugger/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/x86/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/x86/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.DLL.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PublicAssemblies/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PublicAssemblies/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/TestWindow/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Microsoft/TestWindow/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Platform/Debugger/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/CommonExtensions/Platform/Debugger/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/x86/Cosmos.VS.ProjectSystem.EXE.
LOG: Attempting download of new URL file:///C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/Common7/IDE/PrivateAssemblies/DataCollectors/x86/Cosmos.VS.ProjectSystem/Cosmos.VS.ProjectSystem.EXE.
LOG: All probing URLs attempted and failed.
```

**Comment by quajak** [comment]
Thanks for the logs, it seems like actually `Cosmos.VS.ProjectSystem` cannot be found. Can you confirm that the Cosmos Project System extension is installed? (Maybe even try reinstalling it)
Can you also confirm that `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\Extensions\Cosmos\ProjectSystem` exists and contains `Cosmos.VS.ProjectSystems.dll`. 
Could you please also check if the logs earlier on ever mentioned `Cosmos.VS.ProjectSystem`? This seems to be more or a problem with Visual Studio and not the actual build process of Cosmos.

**Comment by tb100** [comment]
Okay, I just reinstalled Cosmos, which ran successfully. `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\Extensions\Cosmos\ProjectSystem` does exist, and it contains `Cosmos.VS.ProjectSystems.dll`. I checked Fusion++ again, and that was the only log for Cosmos (I used the search tool). I also tried to run it again (my operating system project), but it has the same problem as before (`Exception: System.TypeLoadException: Could not load type 'Cosm...`).

**Comment by quajak** [comment]
Can you confirm that in VS both Cosmos extensions are installed?

**Comment by tb100** [comment]
I opened up the NuGet viewer (I think this is what you mean), and it shows `Cosmos.Build`, `Cosmons.Debug.Kernel` and `Cosmos.System2`, along with `NETStandard.Library`. All the Cosmos ones say `Not available in this source`, I'm not sure if that is the problem.

**Comment by quajak** [comment]
Can you select allow Prereleases so that the cosmos ones have a source?

**Comment by tb100** [comment]
I checked the `Include prereleases` box, and restarted Visual Studio. I pressed run, and the `Exception: System.TypeLoadException: Could not load type 'Cosm...` error is gone! I'm getting `Exception: System.Exception: Method GetMBIAddress requires a plug, but none is implemented testOS 1 C:\Users\dev\source\repos\testOS\testOS C:\Users\dev\source\repos\testOS\testOS\IL2CPU 1`, again but I don't think that has anything to do with the `Could not load type 'Cosmos.Core.Multiboot'`. Let me know if this can be closed.
